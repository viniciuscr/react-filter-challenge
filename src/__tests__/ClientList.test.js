import React from "react";
import { mount } from "enzyme";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import ClientList from "../components/clientList";

configure({ adapter: new Adapter() });

describe("Client tests", () => {
  const clientsData = [
    {
      name: "no products",
      id: "5633",
      products: []
    },
    {
      name: "Addf gsd df",
      id: "3423",
      products: [
        {
          product_handle: "lead_management_",
          state: "paused"
        }
      ]
    },
    {
      name: "Pri porter",
      id: "333",
      products: [
        {
          product_handle: "lead_management",
          state: "active"
        },
        {
          product_handle: "marco_polo",
          state: "active"
        }
      ]
    },
    {
      name: "Pitter Bitter",
      id: "3434",
      products: [
        {
          product_handle: "dashboard",
          state: "canceled"
        }
      ]
    }
  ];

  it("should render as expected", () => {
    const wrapper = mount(<ClientList clients={clientsData} />);
    expect(wrapper.html()).toMatchSnapshot();
    wrapper.unmount();
  });
});
