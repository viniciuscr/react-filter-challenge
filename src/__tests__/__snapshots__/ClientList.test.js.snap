// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Client tests should render as expected 1`] = `"<div><div class=\\"shadow-sm h-100  mb-2 border bg-white rounded-sm \\"><div class=\\"client-box\\"><div class=\\"container-fluid\\"><h4>no products <small class=\\"text-muted\\">5633</small></h4><ul class=\\"producsts\\"></ul></div><button class=\\"bt-foward\\"><span class=\\"icon-rl-forward\\"></span></button></div></div><div class=\\"shadow-sm h-100  mb-2 border bg-white rounded-sm \\"><div class=\\"client-box\\"><div class=\\"container-fluid\\"><h4>Addf gsd df <small class=\\"text-muted\\">3423</small></h4><ul class=\\"producsts\\"><li><span class=\\"badge paused\\">paused</span> lead_management_</li></ul></div><button class=\\"bt-foward\\"><span class=\\"icon-rl-forward\\"></span></button></div></div><div class=\\"shadow-sm h-100  mb-2 border bg-white rounded-sm \\"><div class=\\"client-box\\"><div class=\\"container-fluid\\"><h4>Pri porter <small class=\\"text-muted\\">333</small></h4><ul class=\\"producsts\\"><li><span class=\\"badge active\\">active</span> lead_management</li><li><span class=\\"badge active\\">active</span> marco_polo</li></ul></div><button class=\\"bt-foward\\"><span class=\\"icon-rl-forward\\"></span></button></div></div><div class=\\"shadow-sm h-100  mb-2 border bg-white rounded-sm \\"><div class=\\"client-box\\"><div class=\\"container-fluid\\"><h4>Pitter Bitter <small class=\\"text-muted\\">3434</small></h4><ul class=\\"producsts\\"><li><span class=\\"badge canceled\\">canceled</span> dashboard</li></ul></div><button class=\\"bt-foward\\"><span class=\\"icon-rl-forward\\"></span></button></div></div></div>"`;
